### DISCUSSION TITLE
Using the Dynamo DB Session Provider in ECS tasks - with an ECS Task role rather than AWS keys

### DISCUSSION THREAD

**Discussion by gavinconnellotten** [aws/aws-dotnet-session-provider#34] [Q&A] [answered]
Hey all!

We're working to switch to DynamoDB managed session state as part of rearchitecting our ASP .NET 4.7.2 web app. We would like to use the ECS task role for our containers rather than an AWS Profile which uses keys and environment variables for auth. The example configuration for this provider only talks about using Profiles though. Is there a parameter that I'm missing to get the provider to use the ECS task role? Can we leave the **AWSProfileName** parameter out from our config?

At the moment the site is failing to start up with a config error tell me that the _EC2_ role is not available.

Any guidance on this use case would be appreciated!

Thanks,
Gavin

 ```
   <sessionState timeout="10" mode="Custom" customProvider="DynamoDBSessionStoreProvider" xdt:Transform="Replace">
      <providers>
        <add name="DynamoDBSessionStoreProvider"
             type="Amazon.SessionProvider.DynamoDBSessionStateStore"
             AWSProfileName="default"
             Region="ap-southeast-2" />
      </providers>
    </sessionState>
```

**Comment by gavinconnellotten** [comment, marked as answer]
Figured it out! Our IIS app pool wasn't able to see the ECS task credential env vars, didn't have the right permissions to see the AWS created env vars. Switched to using the default app pool and the SDK cannot then see the var and auth works. Leaving this here for anyone else that might experience this in the future!

**Reply by ashishdhingra** [reply]
Thanks for the update.
